src/mpid/ch4/netmod/ofi/ofi_init.c

```diff
@@ -1461,6 +1457,8 @@ static int find_provider(struct fi_info *hints)
 
         fi_freeinfo(prov_list);
     } else {
+        init_global_settings(MPIR_CVAR_OFI_USE_PROVIDER);
```

Comment:
While saving an extra branch, you are moving two code that belongs to the same logic (`init_global_settings`)  far apart, thus less than ideal.

Reply:
I am unsure if I really want to cleanup the init code. The branch was created a while ago. Perhaps I will just drop the cleanup commits. However, even adding the capability back, pt2tp with >=16384bytes still hangs on cori. I need to investigate more.

Reply:
What is the debug capability output after you apply the patch?